doc/decisions/frontend_keyhandling.md

```diff
@@ -0,0 +1,65 @@
+# Frontend Keybindings
+
+## Problem
+
+In the development of PermaplanT, we have identified the need for a robust keybinding system to enhance the user experience. Keybindings are utilized in various parts of the application, and we need to establish a cohesive approach for their implementation.
+
+## Constraints
+
+Our keybinding solution should adhere to the following constraints:
+
+- **Ease of Integration**: The keybinding system should be straightforward to integrate into different components of the application.
+
+- **User Configuration**: We should plan for future functionality that allows users to configure keybindings according to their preferences.
+
+## Decision / Current Solution
+
+We have decided to configure keybindings using a JSON structure that includes scopes and associated keybindings.
+
+Example:
+
+```
+{
+  "global": {
+    "Ctrl+M": "navigateToMap"
+  }
+  "planting": {
+    "Escape": "exitPlantingMode"
+  }
+}
+```
+
+Components that require keybindings can define action handlers for the configured actions. Here's an example of how this can be implemented:
+
+Example:
+
+```
+  const keyHandlerActions : Record<string, () => void> = {
+    exitPlantingMode: () => {
+        exitPlantingMode();
+    }
+  }
+```
+
+We can then add key listeners to specific components, and if a key mapping is configured for the pressed key within the associated scope, the corresponding action can be executed.
+
+## Challenges / To be discussed
+
+### Where do we add key listeners?
+
+While it's easy to add key listeners to focused components like form inputs, it becomes more challenging when dealing with a component like the Konva map. This is because we cannot directly bind keys to the map layer.
+
+Furthermore there are instances where we want to handle keys for map feature even when another component is focused. For example, if a user selects a plant in the right toolbar, pressing the "Escape" key should probably also exit planting mode.
+If we also want to clear a search with the "Escape" key, we have to stop event propagation.
+
+### Interesting npm package
+
+During research following npm package was found which seems to be very promising:
+
+- [react-hotkeys-hook](https://github.com/JohannesKlauss/react-hotkeys-hook/tree/main)
+
+It provides a hook for binding keys in components and also offers scoping abilities.
+
+### Ruled out alternative
+
+- [react-hotkeys](https://www.npmjs.com/package/react-hotkeys): provides a component that receives a keymap and actionhandlers similar to our planned concept and listens to events if a child of the component is in focus. It seams to be very suitable for encapsulating events, however it doesn't really solve the challenge of focus konva layers and although the librarie is widely used the last realase was 4 years ago.
```

Comment:
Put both up to solutions. I agree we can rule out react-hotkeys, as scoping is important from UI perspective.

```suggestion
### React Hotkeys Hook

[react-hotkeys-hook](https://github.com/JohannesKlauss/react-hotkeys-hook/tree/main)

- It provides a hook for binding keys in components and also
- offers scoping abilities.

### React Hotkeys

 [react-hotkeys](https://www.npmjs.com/package/react-hotkeys): provides a component that receives a keymap and actionhandlers similar to our planned concept and listens to events if a child of the component is in focus.

- It seems to be very suitable for encapsulating events, however
- it doesn't really solve the challenge of focus konva layers and
- although the library is widely used the last release was 4 years ago.
```